common/thorhelper/thorcommon.cpp

```diff
@@ -976,11 +988,23 @@ class CRowStreamReader : implements IExtRowStream, public CSimpleInterface
         }
     } crccb;
     
-public:
-    IMPLEMENT_IINTERFACE_USING(CSimpleInterface);
+    inline bool fieldFilterMatch(const void * buffer)
+    {
+        if (actualFilter.numFilterFields())
+        {
+            unsigned numOffsets = actual->getNumVarFields() + 1;
+            size_t * variableOffsets = (size_t *)alloca(numOffsets * sizeof(size_t));
+            RtlRow row(*actual, nullptr, numOffsets, variableOffsets);
```

Comment:
Possibly more efficient - if this object is not accessed by multiple threads - to use a RtlDynRow member variable.